test/bench/synthetic_benchmarks.cpp

```diff
@@ -0,0 +1,240 @@
+}
+
+std::string to_string(const CodeParams& params)
+{
+    return std::string{instr::traits[params.opcode].name} + '/' +
+           static_cast<char>(get_instruction_kind(params.opcode)) +
+           std::to_string(static_cast<int>(params.mode));
+}
+
+/// Generates the EVM benchmarking loop inner code for the given opcode and "mode".
+bytecode generate_loop_inner_code(CodeParams params)
+{
+    const auto [opcode, mode] = params;
+    const auto kind = get_instruction_kind(opcode);
+    switch (mode)
+    {
+    case Mode::min_stack:
+        switch (kind)
+        {
+        case InstructionKind::push:
+            // PUSH1 POP PUSH1 POP ...
+            return stack_limit * (push(opcode, {}) + OP_POP);
+        case InstructionKind::swap:
+        {
+            const auto n = opcode - OP_SWAP1 + 1;
+            // DUP1 SWAP1 SWAP1 ... POP
+            return n * OP_DUP1 + stack_limit * 2 * bytecode{opcode} + n * OP_POP;
+        }
+        case InstructionKind::dup:
+        {
+            const auto n = opcode - OP_DUP1;
+            // DUP1 DUP1 POP DUP1 POP ... POP
+            return n * OP_DUP1 + (stack_limit - n) * (bytecode{opcode} + OP_POP) + n * OP_POP;
+        }
+        case InstructionKind::producer:
+            // CALLER POP CALLER POP ...
+            return stack_limit * (bytecode{opcode} + OP_POP);
+        case InstructionKind::nullop:
+            // JUMPDEST JUMPDEST ...
+            return stack_limit * 2 * bytecode{opcode};
+        case InstructionKind::unop:
+            // DUP1 NOT NOT ... POP
+            return OP_DUP1 + stack_limit * 2 * bytecode{opcode} + OP_POP;
+        case InstructionKind::binop:
+            // DUP1 DUP1 ADD DUP1 ADD DUP1 ADD ... POP
+            return OP_DUP1 + (stack_limit - 1) * (OP_DUP1 + bytecode{opcode}) + OP_POP;
+        default:
+            break;
+        }
+        break;
+    case Mode::full_stack:
+        switch (kind)
+        {
+        case InstructionKind::push:
+            // PUSH1 PUSH1 PUSH1 ... POP POP POP ...
+            return stack_limit * push(opcode, {}) + stack_limit * OP_POP;
+        case InstructionKind::dup:
+        {
+            const auto n = opcode - OP_DUP1;
+            // DUP1 DUP1 POP DUP1 POP ... POP
+            return n * OP_DUP1 + (stack_limit - n) * bytecode{opcode} + stack_limit * OP_POP;
+        }
+        case InstructionKind::producer:
+            // CALLER CALLER ... POP POP ...
+            return stack_limit * opcode + stack_limit * OP_POP;
+        case InstructionKind::binop:
+            // DUP1 DUP1 DUP1 ... ADD ADD ADD ... POP
+            return stack_limit * OP_DUP1 + (stack_limit - 1) * opcode + OP_POP;
+        default:
+            break;
+        }
+        break;
+    }
+
+    return {};
+}
+
+/// Generates a benchmarking loop with given inner code.
+///
+/// This generates do-while loop with 256 iterations and is starts with PUSH1 of 255 as the loop
```

Comment:
```suggestion
/// This generates do-while loop with 256 iterations and it starts with PUSH1 of 255 as the loop
```

Reply:
Ha. I actually miscalculated the loop iterations. It should be 255. Someone please double check.

Reply:
255 looks right.

Reply:
Because I need to re-run the benchmark anyway, should I change the value of this?